File: pkg/ios/locator.go
package ios

import (
	"errors"
	"os"
	"path/filepath"
	"strings"
)

func FindAppPath() (string, error) {
	// Check if there is a .xcodeproj folder in the current directory
	entries, err := os.ReadDir(".")
	if err != nil {
		return "", err
	}

	var projectName string
	for _, entry := range entries {
		if entry.IsDir() && strings.HasSuffix(entry.Name(), ".xcodeproj") {
			projectName = strings.TrimSuffix(entry.Name(), ".xcodeproj")
			break
		}
	}

	if projectName == "" {
		return "", errors.New("❌ No .xcodeproj found. Please run this command from the root of your Xcode project")
	}

	return filepath.Join(".", projectName), nil
}

File: src/packages/ios/locator.ts
import { promises as fs } from 'fs';
import path from 'path';

/**
 * Finds the app path by locating the .xcodeproj directory
 * @returns The path to the app directory (without .xcodeproj extension)
 * @throws Error if no .xcodeproj is found
 */
export async function findAppPath(): Promise<string> {
  // Check if there is a .xcodeproj folder in the current directory
  const entries = await fs.readdir('.', { withFileTypes: true });

  let projectName: string | null = null;

  for (const entry of entries) {
    if (entry.isDirectory() && entry.name.endsWith('.xcodeproj')) {
      projectName = entry.name.replace('.xcodeproj', '');
      break;
    }
  }

  if (!projectName) {
    throw new Error('❌ No .xcodeproj found. Please run this command from the root of your Xcode project');
  }

  return path.join('.', projectName);
}
